src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -235,15 +241,30 @@ private void AnalyzeDirectory(DirectoryInfo directoryPath, ReportFormat reportFo
                         {
                             logger.LogError("\t{failedFile}", failedFile);
                         }
+                        return 6;
                     }
+                    return exitCode;
                 }
             }
             catch (Exception exp)
             {
                 logger.LogError(GetExceptionMessage(exp));
+                return 1;
             }
         }
 
+        private int CalculateExitCode(int exitCode, int res)
+        {
+            if (exitCode == 6 || res == 6)
+                return 6;
+            else if ((exitCode == 5 && res >= 1 && res <= 4) || (res == 5 && exitCode >= 1 && exitCode <= 4))
+                return 6;
+            else if (res == 5)
+                return 5;
+            else
+                return Math.Max(exitCode, res);
```

Comment:
I was not sure if I should do a Max or not. What number should be returned if there are multiple errors?

Reply:
For me it makes sense to return 1 in that case

Reply:
I've seen error codes in other tools where it says "Errors and issues" and "Errors but no issues". Essentially a new code for multiple errors.

Reply:
declining on "Errors but no issues"